numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2449,31 +2444,6 @@ static PySequenceMethods voidtype_as_sequence = {
 };
 
 
-static Py_ssize_t
-gentype_getreadbuf(PyObject *self, Py_ssize_t segment, void **ptrptr)
-{
-    int numbytes;
-    PyArray_Descr *outcode;
-
-    if (segment != 0) {
-        PyErr_SetString(PyExc_SystemError,
-                "Accessing non-existent array segment");
-        return -1;
-    }
-
-    outcode = PyArray_DescrFromScalar(self);
-    numbytes = outcode->elsize;
-    *ptrptr = (void *)scalar_value(self, outcode);
-
-#ifndef Py_UNICODE_WIDE
```

Comment:
Is Py_UNICODE_WIDE no longer a thing? I think I'd prefer we tackle that ifdef in a separate patch, and keep this part for now.

Reply:
That is, let's either:

* Inline the ifdef too, for now
* Put this patch on hold, and make a patch to remove `Py_UNICODE_WIDE` first

Reply:
Just checked - `Py_UNICODE_WIDE` seems to be alive and well, although we'll stop needing when I fixup #15385

Reply:
My reading of the code is that `Py_UNICODE_WIDE` only influences numbytes which is but used by the call site.

Reply:
You're right, thanks for spotting that